### PR TITLE
swap React components for Web Components

### PR DESCRIPTION
**Pull request by rmessina1010**
Swaps React Loading Indicator Component for it's Web Component equivalent in `src/platform/user`; updates corresponding cypress tests and skips unit tests.
## Related issue(s)
[1918](https://app.zenhub.com/workspaces/platform-design-system-5f8de67192551b0012ebb802/issues/gh/department-of-veterans-affairs/vets-design-system-documentation/1918)
 
## Testing done

- Cypress tests (local/CI)
- Unit Tests (local/CI)
- Visual testing

### CODE REVIEW

**Review by radavis — APPROVED** [review]
:+1:

File: src/applications/claims-status/tests/e2e/08.claim-letters-keyboard-only.cypress.spec.js [hunk 1]
```diff
@@ -3,18 +3,21 @@ import claimLetters from './fixtures/mocks/claim-letters/list.json';
...
+    cy.intercept('GET', '/v0/claim_letters', claimLetters.data).as(
+      'lettersLoaded',
+    );
     cy.intercept('GET', '/v0/feature_toggles?*', featureToggleEnabled).as(
       'featureToggleEnabled',
     );
 
     cy.login();
     cy.visit('track-claims/your-claim-letters');
+    // adds .focus() so tests are able to work with the lodading indicator web component.
```
Comment (jerekshoe) [inline src/applications/claims-status/tests/e2e/08.claim-letters-keyboard-only.cypress.spec.js:15]: I might update this to say:
```
// adds .focus() so tests are able to work with the va-loading-indicator web component
```

File: src/applications/claims-status/tests/e2e/08.claim-letters-keyboard-only.cypress.spec.js [hunk 2]
```diff
@@ -3,18 +3,21 @@ import claimLetters from './fixtures/mocks/claim-letters/list.json';
 
 describe('Claim Letters Page', () => {
   beforeEach(() => {
-    cy.intercept('GET', '/v0/claim_letters', claimLetters.data);
+    cy.intercept('GET', '/v0/claim_letters', claimLetters.data).as(
+      'lettersLoaded',
```
Comment (jerekshoe) [inline src/applications/claims-status/tests/e2e/08.claim-letters-keyboard-only.cypress.spec.js:7]: Does this alias make a functional difference? Wondering if it was intentionally added

**Review by jerekshoe — APPROVED** [review]

**Review by randimays — APPROVED** [review]
Yellow Ribbon and Find Forms look good to me! 👍

File: config/cypress.config.js [hunk 3]
```diff
@@ -197,4 +197,13 @@ const cypressConfig = {
...
+module.exports = {
+  ...defineConfig(cypressConfig),
+
+  component: {
+    devServer: {
+      framework: 'react',
+      bundler: 'webpack',
+    },
+  },
+};
```
Comment (adamwhitlock1) [inline config/cypress.config.js:209]: Does this mean that we can use cypress to do their 'component testing' now as well?

**Review by adamwhitlock1 — APPROVED** [review]
lgtm for authenticated experience. Thanks for this!

**Review by simiadebowale — APPROVED** [review]
Noted `vaos` changes. Will create ticket to update test. Thank you

**Comment by adamwhitlock1** [comment]
Im seeing e2e failures with `AssertionError: Timed out retrying after 4000ms: Invalid string length` which im not sure I've seen before.

**Comment by rmessina1010** [comment]
> Im seeing e2e failures with `AssertionError: Timed out retrying after 4000ms: Invalid string length` which im not sure I've seen before.

Yes, that is new. It started, tho I am not 100% it is related to, when I updated src/platform/user/authorization/components/RequiredLoginView.jsx to have a React bound web component. looking into it

**Review by acrollet — APPROVED** [review]
approving for mhv landing page based on @radavis 's approval

**Review by vbahinwillit — COMMENTED** [review]
Changes for VAOS approved

**Review by ryanshaw — APPROVED** [review]

**Review by ataker — APPROVED** [review]

**Pull request merged**